webrender/res/cs_border_segment.glsl

```diff
@@ -102,22 +102,41 @@ vec2 get_outer_corner_scale(int segment) {
     return p;
 }
 
-vec4 mod_color(vec4 color, float f) {
-    return vec4(clamp(color.rgb * f, vec3(0.0), vec3(color.a)), color.a);
+// NOTE(emilio): If you change this algorithm, do the same change
+// in border.rs
+vec4 mod_color(vec4 color, bool is_black, bool lighter) {
+    const float light_black = 0.7;
+    const float dark_black = 0.3;
+
+    const float dark_scale = 0.66666666;
+    const float light_scale = 1.0;
+
+    if (is_black) {
+        if (lighter) {
+            return vec4(vec3(light_black), color.a);
+        }
+        return vec4(vec3(dark_black), color.a);
+    }
+
+    if (lighter) {
+        return vec4(color.rgb * light_scale, color.a);
+    }
+    return vec4(color.rgb * dark_scale, color.a);
 }
 
 vec4[2] get_colors_for_side(vec4 color, int style) {
     vec4 result[2];
-    const vec2 f = vec2(1.3, 0.7);
+
+    bool is_black = color.rgb == vec3(0.0, 0.0, 0.0);
```

Comment:
This may not be portable. IIRC, vector comparison produces vector of booleans. We should be using `dot(color.rgb,color.rgb)` or `all(color.rgb == vec3(0.0, 0.0, 0.0))`

Reply:
I could be wrong but I think that  `some_vec3 == other_vec3` returns a boolean in glsl and that the [equal function](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/equal.xhtml) returns a vector of boolean.

Reply:
Indeed. I just recalled a [driver bug](https://www.opengl.org/discussion_boards/showthread.php/169135-vector-equality) with those, hence my radar fired up the red light :)

Reply:
Also note the relevant record in our own driver bug treasury: https://github.com/servo/webrender/wiki/Driver-issues#1728-1368---shader-compiler-bug-when--is-used-inline-as-function-arg